### ISSUE TITLE
Feature Request

### ISSUE THREAD

**Issue by Coasie** [blokadaorg/blokada#418]
I like the easy to see log and search of the v4 GUI, one addition I'd really like to see in there, is when you click an entry to open it up, {domain forwarded/domain blocked} I'd really like to see which specific app caused the request.

**Comment by agnus73** [comment]
Agree! That would be very heplful!

**Comment by mosaati** [comment]
This is a much appreciated feature

**Comment by lnicola** [comment]
Is this even possible without root?

**Comment by jwsp1** [comment]
Showing which app produced connection XY is an awesome idea. 
Adding to this, it would also be helpful to see all connections per app.

**Comment by kar** [comment]
This is planned. Feel free to "like" this ticket to let me know how many people is interested so we can prioritise.

**Comment by bernd-wechner** [comment]
I'll go one further, if the request has GET parameters visible, would be great to be able to see those and filter on those. Reason being an app may use one common domain name or basic URL for data including ads, and the differentiator is in the GET parameters. May be trickier if they are in the POST parameters (not visible until after the connection is made).

**Comment by kar** [comment]
Yes, but that's another layer of OSI network design, which we don't operate on yet. It's related to 'cosmetic filtering'.

**Comment by lbehal** [comment]
I was looking for this:).
I would like to see which app is using what..It would be also nice to have some additional info/grouping for the blocked domain (advertising/analytics etc.). So I can get some insight about each app. It could also show which apps are using particular tracker.

**Comment by anibalcatheringer** [comment]
Hi, for me this would be my favorite feature from the app, can we see this being on the next to do list?

**Comment by thepatrik3** [comment]
Pls add this feature!

**Comment by R70YNS** [comment]
This feature is absolutely needed. It would really help people identify apps that need removing and/or disabling, which would result in less filtering needed.

**Comment by ps100000** [comment]
**Bad news. Starting with Android 10 access to the info which app created a connection is blocked so until we find a new way to get this information its pointless to implement this feature as it would only support Android < 10.**

**Comment by linuxandria** [comment]
> **Bad news. Starting with Android 10 access to the info which app created a connection is blocked so until we find a new way to get this information its pointless to implement this feature as it would only support Android < 10.**

And yet adguard manages to show that just fine......

**Comment by linuxandria** [comment]
![Screenshot_20200225-090914](https://user-images.githubusercontent.com/45006100/75254648-a9a04380-57ae-11ea-9b14-5616e78193e6.jpg)
![Screenshot_20200225-091045](https://user-images.githubusercontent.com/45006100/75254684-b9b82300-57ae-11ea-8f25-a6b835c55fb2.jpg)

**Comment by ps100000** [comment]
@linuxandria 
If your manufacturer still allows access to /proc/net/ you can call yourself lucky or unlucky depending on how you want to look at it.
In addition adguard might use something like being a device admin to get more info. I don't know for sure as I'm neither a Adguard-developer nor user but just a hobby-dev doing stuff for Blokada.

**Comment by anibalcatheringer** [comment]
@ps100000 I can say that adguard indeed uses device admin but I don't see a problem with that. It could also be implemented in blokada, no?

**Comment by linuxandria** [comment]
@anibalcatheringer you are incorrect. AdGuard never requested device admin details.

**Comment by linuxandria** [comment]
@ps100000 I do however find it odd for me it works on both a stock and custom ROM.
Though why do you even access that? Especially as every packet goes through your tunnel anyhow?

**Comment by linuxandria** [comment]
AdGuard does however request usage access. I bet that's it

**Comment by ps100000** [comment]
> @ps100000 I do however find it odd for me it works on both a stock and custom ROM.
> Though why do you even access that? Especially as every packet goes through your tunnel anyhow?

Because we have to find out which app the package belongs to.
There might be some other way. I'll have another look at some point.

**Comment by jgullickson** [comment]
This would be an incredibly helpful feature. Netguard (https://github.com/M66B/NetGuard) is able to associate requests with specific apps. Perhaps their source code could be a starting point for figuring out how to implement this in Blokada. I'm not a Java developer, but it looks like the AdapterLog.java might be where this functionality is implemented in Netguard. https://github.com/M66B/NetGuard/blob/297c8b2c3030dff502882f2e2ac215a76a6ec3a3/app/src/main/java/eu/faircode/netguard/AdapterLog.java#L329
